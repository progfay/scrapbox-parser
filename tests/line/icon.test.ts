/* global describe it expect */

import '../jest-setup'

describe('icon', () => {
  it('Simple root icon', () => {
    expect('[/icons/+1.icon]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'icon',
            pathType: 'root',
            path: '/icons/+1'
          }
        ]
      }
    ])
  })

  it('Simple relative icon', () => {
    expect('[me.icon]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'icon',
            pathType: 'relative',
            path: 'me'
          }
        ]
      }
    ])
  })

  it('Multiple icons', () => {
    expect('[me.icon*3]').toEqualWhenParsing([
      {
        indent: 0,
        type: 'line',
        nodes: [
          {
            type: 'icon',
            pathType: 'relative',
            path: 'me'
          },
          {
            type: 'icon',
            pathType: 'relative',
            path: 'me'
          },
          {
            type: 'icon',
            pathType: 'relative',
            path: 'me'
          }
        ]
      }
    ])
  })
})
