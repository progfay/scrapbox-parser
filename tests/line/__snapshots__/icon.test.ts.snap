// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon Icon and internal link on same line 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "content": "",
        "href": "Internal link",
        "pathType": "relative",
        "raw": "[Internal link]",
        "type": "link",
      },
      {
        "path": "me",
        "pathType": "relative",
        "raw": "[me.icon]",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Multiple icons 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "path": "me",
        "pathType": "relative",
        "raw": "[me.icon*3]",
        "type": "icon",
      },
      {
        "path": "me",
        "pathType": "relative",
        "raw": "[me.icon*3]",
        "type": "icon",
      },
      {
        "path": "me",
        "pathType": "relative",
        "raw": "[me.icon*3]",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Simple relative icon 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "path": "me",
        "pathType": "relative",
        "raw": "[me.icon]",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;

exports[`icon Simple root icon 1`] = `
[
  {
    "indent": 0,
    "nodes": [
      {
        "path": "/icons/+1",
        "pathType": "root",
        "raw": "[/icons/+1.icon]",
        "type": "icon",
      },
    ],
    "type": "line",
  },
]
`;
